src/madness/world/thread.h

```diff
@@ -1328,12 +1302,15 @@ namespace madness {
             __parsec_schedule(parsec->virtual_processes[0]->execution_streams[0], parsec_task, 0);
             //////////// Parsec Related End ////////////////////
 #elif HAVE_INTEL_TBB
-#ifdef MADNESS_CAN_USE_TBB_PRIORITY
-            if(task->is_high_priority())
-                tbb::task::enqueue(*task, tbb::priority_high);
-            else
-#endif  // MADNESS_CAN_USE_TBB_PRIORITY
-                tbb::task::enqueue(*task);
+//#ifdef MADNESS_CAN_USE_TBB_PRIORITY
+//            if(task->is_high_priority())
+//                tbb::task::enqueue(*task, tbb::priority_high);
+//            else
+//#endif  // MADNESS_CAN_USE_TBB_PRIORITY
+            tbb_arena->enqueue(
+                    [task_p = std::unique_ptr<PoolTaskInterface>(task)](){
+                        task_p->execute();
```

Comment:
It seems you copied a pointer of `PoolTaskInterface` from `task` to `task_p` via `unique_ptr`. Why not just use `task` directly, e.g.
```suggestion
                    [task](){
                        task->execute();
```

Reply:
it has to be deleted as well, which is for `unique_ptr` is used

Reply:
So, you are using ‘unique_ptr’ as an raii guard while an exception is not expected (it will lead to UB if thrown from enqueue task). Does it make sense to specify ‘no except’ for lambda to cause termination in case of exception? I’m addition, add a comment that ‘unique_ptr’ is used to deallocate ‘task’ after execution.